File: dbconnection.py
import pyodbc
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

# Reference the variables
SQL_DATABASE = os.getenv('SQL_DATABASE')
SQL_USERNAME = os.getenv('SQL_USERNAME')
SQL_PASSWORD = os.getenv('SQL_PASSWORD')

# Example usage: print or use these variables in your DB connection logic
print(f"Database: {SQL_DATABASE}")
print(f"Username: {SQL_USERNAME}")
print(f"Password: {SQL_PASSWORD}")

# Connection parameters
server = '192.168.0.7'  # Replace with your Windows PC's IP address
database = SQL_DATABASE  # Use the loaded environment variable
username = SQL_USERNAME      # Use the loaded environment variable
password = SQL_PASSWORD  # Use the loaded environment variable

# Connection string
connection_string = (
    f'DRIVER={{ODBC Driver 18 for SQL Server}};'
    f'SERVER={server};'
    f'DATABASE={database};'
    f'UID={username};'
    f'PWD={password};'
    'Encrypt=yes;'
    'TrustServerCertificate=yes;'
    'Connection Timeout=30;'
)

try:
    # Establish connection
    conn = pyodbc.connect(connection_string)
    print("Connection successful!")
    cursor = conn.cursor()

    # Execute a sample query
    cursor.execute("SELECT * FROM SM.Customers")  # Replace with your table name
    rows = cursor.fetchall()

    # Print results
    for row in rows:
        print(row)

    # Clean up
    cursor.close()
    conn.close()

except Exception as e:
    print(f"Failed to connect or execute query: {e}")

File: dbconnection.js
// honc.dev-compatible DB connection using mssql
import 'dotenv/config';
import sql from 'mssql';

const SQL_DATABASE = process.env.SQL_DATABASE;
const SQL_USERNAME = process.env.SQL_USERNAME;
const SQL_PASSWORD = process.env.SQL_PASSWORD;
const SQL_SERVER = process.env.SQL_SERVER || '192.168.0.7';

console.log(`Database: ${SQL_DATABASE}`);
console.log(`Username: ${SQL_USERNAME}`);
console.log(`Password: ${SQL_PASSWORD}`);
console.log(`Server: ${SQL_SERVER}`);

const config = {
  user: SQL_USERNAME,
  password: SQL_PASSWORD,
  server: SQL_SERVER,
  database: SQL_DATABASE,
  options: {
    encrypt: true,
    trustServerCertificate: true,
  },
  port: 1433,
  connectionTimeout: 30000,
};

export async function handler(event, context) {
  let pool;
  try {
    pool = await sql.connect(config);
    // Query the SM.Customers table
    const result = await pool.request().query('SELECT * FROM [SM].[Customers]');
    console.log(result.recordset);
    return {
      statusCode: 200,
      body: JSON.stringify(result.recordset),
    };
  } catch (err) {
    console.error('MSSQL DB connection error:', err);
    return {
      statusCode: 500,
      body: 'DB connection failed: ' + err.message,
    };
  } finally {
    if (pool) await pool.close();
  }
}

// For local testing only (ESM-compatible)
if (import.meta.url === `file://${process.argv[1]}`) {
  handler({}, {}).then((res) => {
    console.log('Handler result:', res);
  }).catch((err) => {
    console.error('Handler error:', err);
  });
}